cmd/env.go

```diff
@@ -293,6 +332,33 @@ the name of an environment will also update the directory structure in
   ks env set us-west/staging --namespace=staging
 
   # Updates both the name and the URI of the environment 'us-west/staging'.
-  # Updating the name will update the directory structure in 'environments'
-  ks env set us-west/staging --uri=http://example.com --name=us-east/staging`,
+  # Updating the name will update the directory structure in 'environments'.
+  ks env set us-west/staging --uri=http://example.com --name=us-east/staging
+  
+  # Updates URI of the environment 'us-west/staging' based on the server URI
+  # in the context 'staging-west' in your kubeconfig file.
+  ks env set us-west/staging --context=staging-west`,
+}
+
+func commonEnvFlags(flags *pflag.FlagSet) (uri, namespace, context string, err error) {
+	uri, err = flags.GetString(flagEnvURI)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	namespace, err = flags.GetString(flagEnvNamespace)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	context, err = flags.GetString(flagEnvContext)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if len(context) != 0 && len(uri) != 0 {
+		return "", "", "", fmt.Errorf("command cannot take both '%s' and '%s' flags", context, uri)
```

Comment:
Looks like you're using the value of these flags, so an error message would look something like:

```
command cannot take both 'some-ctx' and 'example.com' flags
```

Seems like you probably want something like:

```
command cannot take both '--context' and '--uri' flags
```

Also if we keep this behavior, we probably might want to explain why. Perhaps something like:

```
Flags '--context' and '--uri' are mutually exclusive, because '--context' has a URI. Try setting '--uri', '--namespace' to the desired values.
```